website/cue/reference/components/sources/splunk_hec.cue

```diff
@@ -51,93 +52,8 @@ components: sources: splunk_hec: {
-						required:    false
-						type: uint: {
-							default: 10000000
-							unit:    null
-						}
-					}
-					max_pending_acks_per_channel: {
-						common:      false
-						description: "The maximum number of ack statuses pending query for a single channel. Equivalent to the `max_number_of_acked_requests_pending_query_per_ack_channel` Splunk HEC setting. Minimum of `1`."
-						required:    false
-						type: uint: {
-							default: 1000000
-							unit:    null
-						}
-					}
-					ack_idle_cleanup: {
-						common:      false
-						description: "Whether or not to remove channels after idling for `max_idle_time` seconds. A channel is idling if it is not used for sending data or querying ack statuses."
-						required:    false
-						type: bool: {
-							default: false
-						}
-					}
-					max_idle_time: {
-						common:      false
-						description: "The amount of time a channel is allowed to idle before removal. Channels can potentially idle for longer than this setting but clients should not rely on such behavior. Minimum of `1`."
-						required:    false
-						type: uint: {
-							default: 300
-							unit:    "seconds"
-						}
-					}
-				}
-			}
-		}
-		address: {
-			common:      true
-			description: "The address to accept connections on."
-			required:    false
-			type: string: {
-				default: "0.0.0.0:\(_port)"
-			}
-		}
-		token: {
-			common:      true
-			description: "If supplied, incoming requests must supply this token in the `Authorization` header, just as a client would if it was communicating with the Splunk HEC endpoint directly. If _not_ supplied, the `Authorization` header will be ignored and requests will not be authenticated."
-			required:    false
-			warnings: ["This option has been deprecated, the `valid_tokens` option should be used."]
-			type: string: {
-				default: null
-				examples: ["A94A8FE5CCB19BA61C4C08"]
-			}
-		}
-		valid_tokens: {
-			common:      true
-			description: "If supplied, incoming requests must supply one of these tokens in the `Authorization` header, just as a client would if it was communicating with the Splunk HEC endpoint directly. If _not_ supplied, the `Authorization` header will be ignored and requests will not be authenticated."
-			required:    false
-			type: array: {
-				default: null
-
-				items: type: string: {
-					examples: ["A94A8FE5CCB19BA61C4C08"]
-				}
-			}
-		}
-		store_hec_token: {
-			common:      false
-			description: "When incoming requests contain a Splunk HEC token, if this setting is set to `true`, the token will kept in the event metadata and will be used if the event is sent to a Splunk HEC sink."
-			required:    false
-			type: bool: default: false
-		}
+	configuration: base.components.sources.splunk_hec.configuration & {
+		token: warnings: ["This option has been deprecated, the `valid_tokens` option should be used."]
```

Comment:
I don't think we've consistently been using a `warning` for deprecated options - is that something we should be doing?

Reply:
It wasn't consistent in the old docs either. This is the first time I've come across one that included it as a warning. I quite like it though, and vote we do it throughout. Maybe chat in the stand up about it?

Reply:
👍 to the idea.